lib/active_model/serializer/adapter/json.rb

```diff
@@ -7,37 +7,16 @@ class Json < Adapter
         def serializable_hash(options = nil)
           options ||= {}
           if serializer.respond_to?(:each)
-            @result = serializer.map { |s| FlattenJson.new(s).serializable_hash(options) }
+            @result = serialize_array_without_root(serializer, options)
           else
-            @hash = {}
+            @result = resource_object_for(serializer)
+            @result = add_resource_relationships(
+              @result,
+              serializer,
+              serializer.options[:include])
```

Comment:
I'm curious about this, because here you are on the serializer context, but the included option is an option by association, I thikn this could miss behave if you had multiple associations declaraitons on the serializer using include option, you probably want the include from `serializer.association.<first/whatever/looping>.options`